### PR TITLE
fix: add indexLabel to list and table plugin

### LINKED ISSUE #965: feat(table, list): has possibility to show index number by config

### PR DESCRIPTION
**Pull request by collinlokken**
## Issues related to this change
Closes #965 

image (1)
image (2)
image

### CODE REVIEW

**Comment by roar-larsen** [comment]
Should the index start at 1 instead of 0?

image

File: packages/dm-core/src/components/Table/TableRow/TableRow.tsx [hunk 1]
```diff
@@ -47,12 +48,19 @@ export function TableRow(props: TableRowProps) {
   }
 
   function openItemAsTab() {
+    if (!props.onOpen) {
+      toast.error(
+        'Invalid UiRecipes. The table plugin was not passed an "onOpen()"-function.'
+      )
+      return
+    }
```
Comment (soofstad) [inline packages/dm-core/src/components/Table/TableRow/TableRow.tsx:56]: Should be allowed not to pass an "onOpen" function

**Review by soofstad — CHANGES_REQUESTED** [review]
Not a big fan of this "indexLabel" parameter. 
I would rather see defaults that works good enough.
But if this was decided with PO I guess it's fine

File: packages/dm-core/src/components/Table/TableRow/TableRow.tsx (hunk 1 above)
Comment (collinlokken) [reply packages/dm-core/src/components/Table/TableRow/TableRow.tsx:56]: testet bare kjapt med `if(true) ...` på linje 51
image

**Comment by soofstad** [comment]
Would like your opinion here @l3abak 

In my opinion, this "captionByIndex" and "caption" are misplaced in the uiRecipe for ListPlugin. If they belong there, they belong in any plugin, since any plugin can be passed the "onOpen()"-function.

If they truly are plugin config parameters needed everywhere, I think we should find another solution.

And even better, we can generate these captions with some default rules, that will make for a good enough solution for 90% of the cases.
  - For lists; attribute name of list + index
  - For objects: entity.label ?? entity.name ?? attribute name

**Review by soofstad — APPROVED** [review]

**Pull request merged**